src/M6Web/Bundle/ElasticsearchBundle/Connection/GuzzleConnectionDecorator.php

```diff
@@ -71,10 +75,34 @@ protected function dispatchEvent($method, $uri, $statusCode, $duration)
                 ->setUri($uri)
                 ->setMethod($method)
                 ->setStatusCode($statusCode)
-                ->setDuration($duration);
+                ->setDuration($duration)
+                ->setTook($took);
 
             $this->eventDispatcher->dispatch('m6web.elasticsearch', $event);
         }
     }
 
-}
\ No newline at end of file
+    /**
+     * Extract took from response
+     *
+     * @param array $response
+     *
+     * @return int|null
+     */
+    protected function extractTookFromResponse(array $response)
```

Comment:
It would be nice to avoid code duplication between the two connection decorators